morf-core/src/main/java/org/alfasoftware/morf/upgrade/Upgrade.java

```diff
@@ -183,8 +186,17 @@ public UpgradePath findPath(Schema targetSchema, Collection<Class<? extends Upgr
 
     // Get access to the schema we are starting from
     log.info("Reading current schema");
-    Schema sourceSchema = UpgradeHelper.copySourceSchema(connectionResources, dataSource, exceptionRegexes);
-    SqlDialect dialect = connectionResources.sqlDialect();
+    Schema sourceSchema;
+    List<String> schemaConsistencyStatements;
+    try (SchemaResource databaseSchemaResource = connectionResources.openSchemaResource(dataSource)) {
+      sourceSchema = SchemaUtils.copy(databaseSchemaResource, exceptionRegexes);
+      schemaConsistencyStatements = connectionResources.sqlDialect().getSchemaConsistencyStatements(databaseSchemaResource);

```

Comment:
pretty minor but could just use the existing dialect variable rather than calling connectionResources.sqlDialect()